pallets/parachain-staking/src/lib.rs

```diff
@@ -1730,9 +1731,7 @@ pub mod pallet {
 				));
 			}
 			let mint = |amt: BalanceOf<T>, to: T::AccountId| {
-				if let Ok(imb) = T::Currency::deposit_into_existing(&to, amt) {
```

Comment:
If `deposit_into_existing()` were to somehow fail here, we would end up emitting an event when no deposit was made. That's functionally all that changes here.

Side note: this function could fail if the account doesn't exist. Is it possible that the account could be reaped (due to existential deposit) if `existential_deposit` > `staking_minimum`? I guess in that case we would effectively burn the rewards...

Reply:
> If deposit_into_existing() were to somehow fail here, we would end up emitting an event when no deposit was made.
The existing code reads `if T::Currency::deposit_into_existing` succeeds => returns `Ok(imbalance)` then emit the `Rewarded` event. If the transfer fails, the event is NOT emitted in the current code.

With the changes, the event is emitted even if the transfer fails; we do not want this.

Reply:
The `deposit_into_existing` function fails if the account doesn't exist (`DeadAccount`). In the existing code, this would return an `Err()` from `T::Currency::deposit_existing` which would mean the event is not emitted.
```rust
/// Deposit some `value` into the free balance of an existing target account `who`.
	///
	/// Is a no-op if the `value` to be deposited is zero.
	fn deposit_into_existing(
		who: &T::AccountId,
		value: Self::Balance,
	) -> Result<Self::PositiveImbalance, DispatchError> {
		if value.is_zero() {
			return Ok(PositiveImbalance::zero())
		}

		Self::try_mutate_account(
			who,
			|account, is_new| -> Result<Self::PositiveImbalance, DispatchError> {
				ensure!(!is_new, Error::<T, I>::DeadAccount);
				account.free = account.free.checked_add(&value).ok_or(ArithmeticError::Overflow)?;
				Ok(PositiveImbalance::new(value))
			},
		)
	}
```

With the changes, the event would be emitted even if the transfer fails for any reason. The existing code only emits the event if the transfer succeeds and it returns `Ok(amt_transferred)`.

Reply:
Yes, this is my conclusion as well. Properly handling this in my PR would complicate things a lot, I think we would need to keep a map of `recipient -> (collator, reward_amount)` to properly emit the event only after a successful payout.

This is really an edge case, though, so one thing we could do is require that the minimum nomination be more than the existential_deposit in which case this could never happen...

Reply:
Oh, and it's not just about complicating the code -- it's about the extra runtime overhead... this extra complexity would mean less room for extrinsics...